modules/custom/az_publication/az_publication.install

```diff
@@ -83,3 +85,172 @@ function az_publication_update_1020702() {
+    'report' => 'az_report',
+    'review' => 'az_review',
+    'review-book' => 'az_review_book',
+    'software' => 'az_software',
+    'song' => 'az_song',
+    'speech' => 'az_speech',
+    'standard' => 'az_standard',
+    'thesis' => 'az_thesis',
+    'treaty' => 'az_treaty',
+    'webpage' => 'az_webpage',
+  ];
+
+  // Process nodes in batches.
+  // Number of nodes to process per batch.
+  $limit = 50;
+  $nids_slice = array_slice($sandbox['nids'], $sandbox['progress'], $limit);
+  $nodes = Node::loadMultiple($nids_slice);
+  $time = \Drupal::service('datetime.time');
+
+  foreach ($nodes as $node) {
+    if ($node->hasField('field_az_publication_type') && !$node->get('field_az_publication_type')->isEmpty()) {
+      $old_value = $node->get('field_az_publication_type')->value;
+      $new_value = $mapping[$old_value];
+
+      // Map and update the new entity reference field.
+      if (isset($mapping[$old_value])) {
+        $node->set('field_az_publication_mapped_type', ['target_id' => $mapping[$old_value]]);
+        $node->setNewRevision(TRUE);
+        $node->isDefaultRevision(TRUE);
+        // Construct a detailed message.
+        $revision_log_message = sprintf(
+          "Publication type updated from '%s' to '%s' as part of data migration to a customizable type reference field. For more information, see the issue at https://github.com/az-digital/az_quickstart/issues/2547.",
+          $old_value,
+          $new_value
+        );
+        $node->setRevisionLogMessage($revision_log_message);
+        $node->setRevisionCreationTime($time->getRequestTime());
+        $node->setRevisionUserId(1);
+        $node->save();
+      }
+    }
+
+    $sandbox['progress']++;
+    $sandbox['current_node'] = $node->id();
+  }
+
+  // Inform the batch API about the progress.
+  $sandbox['#finished'] = $sandbox['progress'] / $sandbox['max'];
+}
+
+/**
+ * Helper function to return batch info.
+ */
+function az_publication_batch_info() {
+  return [
+    'title' => t('Updating AZ Publication Nodes...'),
+    'operations' => [
+      ['az_publication_update_1020902', []],
+    ],
+    'finished' => 'az_publication_batch_finished',
+  ];
+}
+
+/**
+ * Finished callback for the batch.
+ */
+function az_publication_batch_finished($success, $results, $operations) {
+  $messenger = \Drupal::messenger();
+
+  if ($success) {
+    $message = \Drupal::translation()->formatPlural(count($results),
+      'One publication successfully processed.', '@count publications successfully processed.');
+  }
+  else {
+    // Handle the case where the batch did not complete successfully.
+    // $operations contains the operations that remained unprocessed.
+    $error_operation = reset($operations);
+    $message = t('An error occurred while processing %error_operation with arguments: @arguments', [
+      '%error_operation' => $error_operation[0],
+      '@arguments' => print_r($error_operation[1], TRUE),
+    ]);
+  }
+
+  $messenger->addMessage($message);
+}
```

Comment:
For some reason I am not seeing messages during update.
![Uploading image.png…]()